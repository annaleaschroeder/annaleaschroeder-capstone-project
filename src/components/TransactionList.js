import React from 'react'
import Transaction from './Transaction'

export default function TransactionList({ transactions }) {
  console.log(transactions)

  return (
    <ul>
      {transactions.map(({ timestamp, transactionvalue, id }) => (
        <li key={id}>
          <Transaction
            createdAt={timestamp}
            newTransaction={transactionvalue}
          />
        </li>
      ))}
    </ul>
  )
}
