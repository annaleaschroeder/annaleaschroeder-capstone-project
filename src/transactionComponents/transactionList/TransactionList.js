import React from 'react'
import Transaction from '../singleTransaction/Transaction'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

TransactionList.propTypes = {
  transactions: PropTypes.arrayOf(PropTypes.object).isRequired,
}

export default function TransactionList({ deleteTransaction, transactions }) {
  return (
    <StyledList data-cy="transactionsList">
      {transactions.map(({ timestamp, value, id, type, notes, tag }) => (
        <li key={id}>
          <Transaction
            deleteTransaction={(event) => {
              event.stopPropagation()
              if (window.confirm('Are you sure?')) {
                deleteTransaction(id)
              }
            }}
            createdAt={timestamp}
            newTransaction={value}
            type={type}
            notes={notes}
            tag={tag}
            id={id}
          />
        </li>
      ))}
    </StyledList>
  )
}

const StyledList = styled.ul`
  list-style: none;
  padding-left: 0;
`
