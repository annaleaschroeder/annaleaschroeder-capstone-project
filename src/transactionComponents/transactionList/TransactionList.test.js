import React from 'react'
import TransactionList from './TransactionList'
import renderer from 'react-test-renderer'

describe('Lists all incoming Transactions', () => {
  it('renders correctly', () => {
    const transaction = [
      { id: '1', createdAt: '9.1.2020', value: '- 15,99 â‚¬', tag: 'food' },
    ]
    const tree = renderer.create(<TransactionList transactions={transaction} />)
    expect(tree).toMatchSnapshot()
  })
})
