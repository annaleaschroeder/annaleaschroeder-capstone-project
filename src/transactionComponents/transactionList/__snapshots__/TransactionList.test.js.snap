// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lists all incoming Transactions renders correctly 1`] = `
<ul
  className="TransactionList__StyledList-eypxhw-0 nuLIe"
  data-cy="transactionsList"
>
  <li>
    <span
      className="Transaction__TimestampStyled-sc-1c8pu3h-0 lnVrnB"
    >
      9.1.2020
    </span>
    <section
      className="Transaction__StyledTransaction-sc-1c8pu3h-1 eibPvK"
      data-cy="transaction-1"
      onClick={[Function]}
    >
      <div
        className="Transaction__PositionDeleteButton-sc-1c8pu3h-2 cTuiyF"
      >
        <button
          className="DeleteButton__StyledButton-nra254-0 dOmPfU"
          id="delete"
          onClick={[Function]}
        >
          <svg
            fill="grey"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </button>
      </div>
      <div
        className="Transaction__PositionEditButton-sc-1c8pu3h-3 BoDNw"
      >
        <button
          className="EditButton__StyledButton-sc-1rgdk1k-0 caOfdr"
          id="edit"
          onClick={[Function]}
        >
          <svg
            fill="grey"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
          </svg>
        </button>
      </div>
      <span
        className="Transaction__TagContainer-sc-1c8pu3h-4 fRaOQb"
      >
        <div
          className="Transaction__TagStyled-sc-1c8pu3h-5 iLUawe food"
        >
          food
        </div>
      </span>
      <span
        className="Transaction__ValueStyled-sc-1c8pu3h-6 hhIzvF spending"
      >
        -â‚¬15.99
      </span>
    </section>
  </li>
</ul>
`;
